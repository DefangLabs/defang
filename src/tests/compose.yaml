name: tests
services:
  dfnx:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        DNS: "${COMPOSE_PROJECT_NAME}-dfnx.prod1.defang.dev"
    env_file:
      - fileName.env
    deploy:
      resources:
        limits:
          cpus: '0.50'
          memory: 512M
        reservations:
          cpus: '0.25'
          memory: 256M
    ports:
      - target: 80
        mode: ingress
      - target: 1234
        # mode: host
    secrets:
      - dummy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
      # disable: true

  # dfnx:
  #   build:
  #     context: .
      # dockerfile: Dockerfile.dfn
    # ports:
      # - 80

  echo:
    image: ealen/echo-server
    ports:
      - target: 80
        mode: ingress
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
    # domainname: echotest.gnafed.click
    profiles:
      - donotstart

secrets:
  dummy:
    external: true
    name: dummy
