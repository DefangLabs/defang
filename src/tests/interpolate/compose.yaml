include: ["compose.${NODE_ENV-dev}.yaml"]
services:
  interpolate:
    image: alpine
    environment:
      - NAME=$COMPOSE_PROJECT_NAME
      - BRACED=${COMPOSE_PROJECT_NAME}
      - DB=postgres://user:$POSTGRES_PASSWORD@db:5432/db
      - ${COMPOSE_PROJECT_NAME}=value
      - NOP=abc$$def # FIXME: this should not get resolved in CD
      - NOP_BRACED=abc$${def} # FIXME: this should not get resolved in CD
      - NODE_ENV=${NODE_ENV}
      - PORT=${PORT:-8080}
