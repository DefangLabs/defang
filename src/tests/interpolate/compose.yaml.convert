[
  {
    "name": "interpolate",
    "image": "alpine",
    "platform": 2,
    "internal": true,
    "environment": {
      "BRACED": "interpolate",
      "DB": "postgres://user:${POSTGRES_PASSWORD}@db:5432/db",
      "NAME": "interpolate",
      "NOP": "abc$def",
      "NOP_BRACED": "abc${def}",
      "interpolate": "value"
    },
    "networks": 1
  }
]