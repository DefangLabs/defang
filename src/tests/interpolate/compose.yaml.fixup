[
  {
    "command": null,
    "entrypoint": null,
    "environment": {
      "BRACED": "interpolate",
      "DB": "postgres://user:${POSTGRES_PASSWORD}@db:5432/db",
      "NAME": "interpolate",
      "NODE_ENV": "${NODE_ENV}",
      "NOP": "abc$def",
      "NOP_BRACED": "abc${def}",
      "PORT": "8080",
      "interpolate": "value"
    },
    "image": "alpine",
    "networks": {
      "default": null
    }
  }
]