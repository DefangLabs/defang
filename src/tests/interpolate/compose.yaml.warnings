 ! service "interpolate": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)
 ! service "interpolate": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors