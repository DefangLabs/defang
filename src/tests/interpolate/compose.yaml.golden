name: interpolate
services:
  interpolate:
    environment:
      BRACED: interpolate
      DB: postgres://user:${POSTGRES_PASSWORD}@db:5432/db
      NAME: interpolate
      NOP: abc$def
      NOP_BRACED: abc${def}
      interpolate: value
    image: alpine
    networks:
      default: null
networks:
  default:
    name: interpolate_default
